content/blog/en/version-6-dojo.md

```diff
@@ -0,0 +1,295 @@
+
+Configuration required prior to version 6 for compiling a Dojo widget to a custom element:
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+> src/MyWidget
+```tsx
+import { WidgetBase } from '@dojo/framework/core/WidgetBase';
+import { customElement } from '@dojo/framework/core/decorators/customElement';
+import { CustomElementChildType } from '@dojo/framework/core/registerCustomElement';
+
+interface MyWidgetProperties {
+  value: string;
+  disabled: boolean;
+  onInput: (input: value) => void;
+}
+
+@customElement<MyWidgetProperties>({
+	tag: 'dojo-my-widget',
+	childType: CustomElementChildType.TEXT,
+	properties: ['disabled'],
+	attributes: ['value'],
+	events: [
+		'onInput'
+	]
+})
+class MyWidget extends WidgetBase<MyWidgetProperties> {
+  protected render() {
+    // rendering
+  }
+}
+```
+
+In Dojo version 6, this is substantially simplified. Now the Dojo build tool's intelligent configuration of custom elements only requires the `.dojorc` entry and will automatically include new, changed or removed properties!
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+## BTR and Dojo Block Improvements
+
+In version 5 of Dojo, we announced Dojo Blocks, a feature leveraging build time rendering (BTR) that brought the world of static site generation to Dojo. Since then we have been working on improving the experience including more intelligent block bundling, dynamic path registration and a full static mode. Building static and progressive websites has never been easier with Dojo.
+
+Review the [example static blog site with Dojo on CodeSandbox](https://codesandbox.io/s/my-first-blog-bywnn) to learn more about BTR and Dojo Block improvements.
+
+## Widget Library Build
+
+A long awaited and highly requested feature for Dojo has been support for building Dojo widget libraries using the `@dojo/cli-build-widget` command. As part of Dojo 6 we're excited to include a library target for the first time. We are now using this to build the `@dojo/widgets` library and are very excited to see more Dojo widget libraries popping up throughout the community ❤️.
+
+To build your widgets using `@dojo/cli-build-widget`, list your widgets in the `widgets` section of the `.dojorc` and run the build using the `--lib` option. The resulting build output is ready to be packaged and consumed by other Dojo applications.
+
+> .dojorc
+```json
+{
+  "build-widget": {
+    "widgets": [
+      "src/MyWidget"
+    ]
+  }
+}
+```
+
+> terminal
+```shell
+dojo build widget --lib
+```
+
+## Faster Development Builds
```

Comment:
Can we provide any info on how this is done? Are we intelligently detecting which files to recompile? Maybe it's too complex to mention here, but a bit of detail might be helpful.

Reply:
We don't tend to go into much detail around the implementation as these things can change, so probably best to keep it light on that front.